datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -0,0 +1,159 @@
+import logging
+
+from datagateway_api.src.common.base_query_filter_factory import QueryFilterFactory
+from datagateway_api.src.common.exceptions import FilterError
+from datagateway_api.src.search_api.filters import (
+    SearchAPIIncludeFilter,
+    SearchAPILimitFilter,
+    SearchAPISkipFilter,
+    SearchAPIWhereFilter,
+)
+
+log = logging.getLogger()
+
+
+class SearchAPIQueryFilterFactory(QueryFilterFactory):
+    @staticmethod
+    def get_query_filter(request_filter, entity_name=None):
```

Comment:
We need to think about how we get `entity_name` for an endpoint. This is easy for when the Text operator is part of an _INCLUDE_ filter because the entity_name is essentially the value specified in the `relation` field. However, what about when there is only a _WHERE_ filter that has a Text operator?

Reply:
I've addressed this for you in 1d24021ce32fe7fa9843de7c438aa6713d4eb44e. You can now use `entity_name` passed into `get_query_filter()` to know which entity name is used.

Reply:
Thanks for that!